src/validate_deepstream_vs_onnx.py

```
@@ -0,0 +1,265 @@
+
+                            if np_dtype == np.uint16:
+                                # view to float16 then cast to float32 for comparison
+                                arr = arr.view(np.float16).astype(np.float32, copy=False)
+                            elif np_dtype in (np.int8, np.int32):
+                                arr = arr.astype(np.float32, copy=False)
+
+                            result["tensor"] = arr.copy()
+
+                    l_user = l_user.next
+                l_frame = l_frame.next
+
+        except Exception as e:
+            print(f"[WARN] Could not extract tensor meta in probe: {e}")
+
+        return Gst.PadProbeReturn.OK
+
+    # Attach probe on nvinfer src to capture tensor meta
+    nvinfer_src_pad = nvinfer.get_static_pad("src")
+    if nvinfer_src_pad is None:
+        print("[ERROR] Could not get nvinfer src pad")
+        return None
+    nvinfer_src_pad.add_probe(Gst.PadProbeType.BUFFER, tensor_extraction_probe, None)
+
+    pipeline.set_state(Gst.State.PLAYING)
+    loop = GLib.MainLoop()
+    Timer(5, loop.quit).start()
+    loop.run()
+    pipeline.set_state(Gst.State.NULL)
+
+    return result.get("tensor", None)
+
+
+def main(config_file, test_dir, model_name, onnx_dir, save_ds_preds=None, fail_on_missing_onnx=False):
+    test_images = sorted(
+        list(Path(test_dir).glob("*.jpg"))
+        + list(Path(test_dir).glob("*.jpeg"))
+        + list(Path(test_dir).glob("*.png"))
+    )
+    if not test_images:
+        print(f"[WARN] No images found in {test_dir}")
+
+    onnx_dir = Path(onnx_dir)
+    save_dir = Path(save_ds_preds) if save_ds_preds else None
+
+    for img in test_images:
+        print(f"[INFO] Running DeepStream on {img.name}")
+        ds_tensor = run_deepstream_inference(config_file, img)
+        if ds_tensor is None:
+            print(f"[ERROR] No tensor returned from DeepStream for {img.name} — skipping comparison")
```

if you have an error, then please throw an error, not `continue`